java_src/foam/dao/FileJournal.java

```diff
@@ -0,0 +1,102 @@
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.core.Journal;
+import foam.lib.json.JournalParser;
+import foam.lib.json.Outputter;
+
+import java.io.*;
+
+/**
+ * Created by carlos on 2017-05-22.
+ */
+public class FileJournal implements Journal {
+
+    protected BufferedWriter bw;
+    protected BufferedReader br;
+    protected File file;
+
+    public FileJournal(String filename) throws IOException {
+
+        this.file = new File(filename);
+
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+
+        this.bw = new BufferedWriter(new FileWriter(file, true));
+        this.br = new BufferedReader(new FileReader(file));
+    }
+
+    /**
+     * Persists data into the File System.
+     *
+     * @param obj
+     * @param sub
+     */
+    @Override
+    public void put(FObject obj, Detachable sub) {
+        try {
+            Outputter outputter = new Outputter();
+            this.bw.write("p(foam.json.parse(" + outputter.stringify(obj) + "))");
```

Comment:
The javascript JournaledDAO uses `eval();` to load it. That's why it's stored that way.